### PR TITLE
ci(build-deb): Perform lintian checks on sources and binary packages

### PR DESCRIPTION
**Pull request by 3v1n0**
UDENG-5415

### CODE REVIEW

**Comment by 3v1n0** [comment]
FYI (@CarlosNihelton, @ashuntu) in the wsl pro client with this we only have two minor warnings:

On sources:
```
 W: wsl-pro-service source: superfluous-file-pattern vendor/go.uber.org/atomic/* [debian/copyright:100]
```

On binaries:
```
  W: wsl-pro-service: hardening-no-pie [usr/libexec/wsl-pro-service]
  P: wsl-pro-service: spelling-error-in-copyright Unknwon Unknown
```

But none of these leads to a build failure.

File: gh-actions/common/build-debian/action.yml [hunk 1]
```diff
@@ -17,6 +17,10 @@ inputs:
     required: false
     # FIXME: this should default to '', but we don't want to break job depending on us for now
     default: 'ca-certificates git'
+  lintian:
+    required: false
+    description: Arguments to pass to lintian, if any. Set to false to skip the lintian check.
```
Comment (didrocks) [inline gh-actions/common/build-debian/action.yml:22]: and why not `skip`?

File: gh-actions/common/build-debian/action.yml [hunk 2]
```diff
@@ -160,6 +164,40 @@ runs:
         path: ${{ env.SOURCE_OUTPUT_DIR }}/
         if-no-files-found: error
 
+    - name: Run lintian on sources
```
Comment (didrocks) [inline gh-actions/common/build-debian/action.yml:167]: I would be good to explain why you want to run lintian in a container and not just installing it as part of dev package. (I do see you create an user, but I’m blanking on why)

File: gh-actions/common/build-debian/action.yml [hunk 3]
```diff
@@ -160,6 +164,40 @@ runs:
...
+
+          echo "::group::Create test user"
+          adduser --disabled-password --gecos "" tester
+          chown tester:tester . -R
+          echo "::endgroup::"
+
+          echo "::group::Run lintian"
+          runuser -u tester -- \
+            lintian --pedantic --fail-on error ${{ inputs.lintian }} -- \
+              "${{ env.PKG_NAME }}_${{ env.PKG_VERSION }}_source".changes
```
Comment (didrocks) [inline gh-actions/common/build-debian/action.yml:198]: Do you think we can annotate the Github Action CI build itself with the debug:warnings/error from it? If you want to see how to write something there, you can use something like https://github.com/canonical/desktop-engineering/blob/main/gh-actions/common/print-summary/action.yml#L33. Or you can format special warnings to be printed.

**Review by didrocks — CHANGES_REQUESTED** [review]
2 suggestions and one remark that needs clarification! Otherwise great! :)

File: gh-actions/common/build-debian/action.yml (hunk 3 above)
Comment (3v1n0) [reply gh-actions/common/build-debian/action.yml:198]: Yeah, you're reading my mind on that.... I was in fact actually working on a script to generate that right now :-D

File: gh-actions/common/build-debian/action.yml (hunk 2 above)
Comment (3v1n0) [reply gh-actions/common/build-debian/action.yml:167]: Well, it brings a lot of dependencies in, so I don't want end up in creating false positives by being able to build a source package when the required -dev dependencies are not yet installed.

File: gh-actions/common/build-debian/action.yml (hunk 1 above)
Comment (3v1n0) [reply gh-actions/common/build-debian/action.yml:22]: ah, `skip` that's fine too, I think I was basing this choice on some older PR that was suggested

File: gh-actions/common/build-debian/action.yml (hunk 2 above)
Comment (didrocks) [reply gh-actions/common/build-debian/action.yml:167]: mind adding a comment for it?

**Comment by 3v1n0** [comment]
| Result | Pacakge | Tag | Details |
|:--:|:--:|--|--|
| 🟠 |foo-pkg |`absolute-symbolic-link-target-in-source` | |
| 🟠 |foo-pkg |`absolute-symlink-in-top-level-folder` | |
| 🟡 |foo-pkg |`acute-accent-in-manual-page` | |
| 🟡 |foo-pkg |`add-component-copyright` | |
| 🟠 |foo-pkg |`adduser-with-home-var-run` | |
| 🟡 |foo-pkg |`adopted-extended-field` | |
| 🔴 |foo-pkg |`alternates-not-allowed` | |
| 🟠 |foo-pkg |`alternatively-build-depends-on-python-sphinx-and-python3-sphinx` | |
| 🟠 |foo-pkg |`ambiguous-paragraph-in-dep5-copyright` | |
| 🟠 |foo-pkg |`ancient-libtool` | |
| 🟠 |foo-pkg |`ancient-python-version-field` | |
| 🟠 |foo-pkg |`ansi-escape` | |
| 🟡 |foo-pkg |`anticipated-repack-count` | |
| 🔴 |foo-pkg |`apache2-configuration-files-need-conf-suffix` | |
| 🟠 |foo-pkg |`apache2-deprecated-auth-config` | |
| 🔴 |foo-pkg |`apache2-module-does-not-depend-on-apache2-api` | |
| ⚪ |foo-pkg |`wiki-copyright-format-uri` | |
| ⚪ |hello |`copyright-refers-to-symlink-license` |usr/share/common-licenses/GPL |
| 🟠 |wsl-pro-service |`hardening-no-pie` |[usr/libexec/wsl-pro-service] |
| ⚪ |wsl-pro-service |`spelling-error-in-copyright` |Unknwon Unknown |

File: gh-actions/common/build-debian/action.yml (hunk 2 above)
Comment (3v1n0) [reply gh-actions/common/build-debian/action.yml:185]: Ok done

**Review by didrocks — APPROVED** [review]
Great work! That almost tempt me to get some lintian bugs on my packages to see this beautiful summary! :)

**Pull request merged**